crates/orchestrator/src/jobs/mod.rs

```diff
@@ -124,34 +126,69 @@ pub trait Job: Send + Sync {
 }
 
 /// Creates the job in the DB in the created state and adds it to the process queue
+#[tracing::instrument(name = "general_create_job", skip(config))]
 pub async fn create_job(
     job_type: JobType,
     internal_id: String,
     metadata: HashMap<String, String>,
     config: Arc<Config>,
 ) -> Result<(), JobError> {
+    let block_gauge = telemetry::global_meter()
+        .f64_gauge(format!("{:?}{}", OTEL_SERVICE_NAME, "_block_state"))
+        .with_description("A gauge to show block state at given time")
+        .with_unit("block")
+        .init();
+
+    tracing::info!("CREATE JOB CALLED for {:?} and type {:?}", internal_id, job_type);
+
     let existing_job = config
         .database()
         .get_job_by_internal_id_and_type(internal_id.as_str(), &job_type)
         .await
         .map_err(|e| JobError::Other(OtherError(e)))?;
+
     if existing_job.is_some() {
+        tracing::info!("CREATE JOB SKIPPED for {:?} and type {:?}", internal_id, job_type);
         return Err(JobError::JobAlreadyExists { internal_id, job_type });
     }
 
     let job_handler = factory::get_job_handler(&job_type).await;
-    let job_item = job_handler.create_job(config.clone(), internal_id, metadata).await?;
+    let job_item = job_handler.create_job(config.clone(), internal_id.clone(), metadata).await?;
     config.database().create_job(job_item.clone()).await.map_err(|e| JobError::Other(OtherError(e)))?;
 
     add_job_to_process_queue(job_item.id, config.clone()).await.map_err(|e| JobError::Other(OtherError(e)))?;
+
+    tracing::info!("JOB CREATED for {:?} and type {:?} with Id {:?}", internal_id, job_type, job_item.id);
+
+    let attributes = [
+        KeyValue::new("job_type", format!("{:?}", job_type)),
+        KeyValue::new("type", "create_job"),
+        KeyValue::new("job", format!("{:?}", job_item)),
+    ];
+
+    block_gauge.record(internal_id.parse::<f64>().unwrap(), &attributes);
+
     Ok(())
 }
 
 /// Processes the job, increments the process attempt count and updates the status of the job in the
 /// DB. It then adds the job to the verification queue.
+#[tracing::instrument(name = "general_process_job", skip(config), fields(job, job_type, internal_id))]
 pub async fn process_job(id: Uuid, config: Arc<Config>) -> Result<(), JobError> {
+    let block_gauge = telemetry::global_meter()
+        .f64_gauge(format!("{:?}{}", OTEL_SERVICE_NAME, "_block_state"))
+        .with_description("A gauge to show block state at given time")
+        .with_unit("block")
+        .init();
+
     let mut job = get_job(id, config.clone()).await?;
 
+    tracing::Span::current().record("job", format!("{:?}", job.clone()));
+    tracing::Span::current().record("job_type", format!("{:?}", job.job_type.clone()));
+    tracing::Span::current().record("internal_id", job.internal_id.clone());
```

Comment:
we did a different method above, should we use the same thing? any way to generalise to a function or maybe create a macro?

Reply:
The `attributes` are being used to record with `metrics`,
to record any value within `tokio traces` this is the best way I found [here](https://docs.rs/tracing/0.1.40/tracing/attr.instrument.html#:~:text=tracing%3A%3ASpan%3A%3Acurrent().record(%22result%22%2C%20%26result)%3B).

Reply:
a `trace_record!` macros seems pretty appropriate to me here, I can use it like

```rust
trace_record_attribute!("job", format!("{:?}", job.clone()));
```

Reply:
**Discussed offline.**
Keeping as it is, since this is minimal changes.